### PR TITLE
Add append builder for Bytes object

### PR DESCRIPTION
**Pull request by LLITCHEV**
**Description**:

**Related issue(s)**:

Fixes # https://github.com/hashgraph/pbj/issues/34
**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by jasperpotts** [comment]
Not sure the approach is quite right here. To start with can we add a two methods to Bytes:
```
public Bytes append(Bytes data) { ...}
public Bytes append(RandomAccessData data) { ...}
```
They return a new Bytes object that contains the bytes of this bytes object followed by all the bytes of the supplied object. The one that takes a Bytes object can have an efficient implementation that does zero copies of data by having an implementation of Bytes with two inner Bytes objects and direct referencing their data. Because Bytes is immutable it will work.

**Comment by LLITCHEV** [comment]
I see... I think we can keep the Builder class for now. It is doing copy and it has bunch of convenient methods and it is tested. I'm implementing the Bytes way with no copying. One thing is that you can append multiple Bytes objects to a Bytes objects. This is what I intend to support. Having a List of Bytes in the Bytes and modifying all the methods to deal with it. Not sure how this will make the reading of primitives (VarInt, etc, especially with unsafe buffers). I suspect that delegating to the appended Bytes might introduce some inefficiencies. But lets measure things.

File: pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/Bytes.java [hunk 1]
```diff
@@ -556,4 +556,41 @@ private static Comparator<Bytes> valueSorter(@NonNull final Comparator<Byte> byt
...
+
+    /**
+     * Appends a {@link Bytes} object to this {@link Bytes} object, producing a new immutable {link Bytes} object.
+     * @param bytes2 The second {@link Bytes} object to append.
+     * @return A new {link Bytes} object containing the concatenated bytes and b.
+     * @throws BufferUnderflowException if the buffer is empty
+     * @throws IndexOutOfBoundsException If the given {@code offset} is negative or not less than Bytes.length()
+     */
+    @NonNull
+    public Bytes appendBytes(@NonNull final Bytes bytes) {
```
Comment (jasperpotts) [inline pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/Bytes.java:568]: can the name of both methods just be `append()` with just parameter types different

**Review by jasperpotts — CHANGES_REQUESTED** [review]

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/BytesBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+final class BytesBuilderTest {
```
[the remark this answers is not in the document]
Comment (LLITCHEV) [reply pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/BytesBuilderTest.java:31]: Ops, I thought I removed it. Done.

File: pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/Bytes.java (hunk 1 above)
Comment (LLITCHEV) [reply pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/Bytes.java:568]: Done.

File: pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/Bytes.java [hunk 3]
```diff
@@ -556,4 +556,41 @@ private static Comparator<Bytes> valueSorter(@NonNull final Comparator<Byte> byt
...
+        // The length field of Bytes is int. The length(0 returns always an int,
```
[the remark this answers is not in the document]
Comment (LLITCHEV) [reply pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/Bytes.java:569]: Fixed.

**Review by jasperpotts — APPROVED** [review]

**Pull request merged**